File: 100-Days-Series/Day 43/RotateMatrix90d.java
public class RotateMatrix90d {
    // Function to rotate a matrix by 90 degrees clockwise
    public static void getRotate(int matrix[][]){
        int n=matrix.length;
        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                int temp=matrix[i][j];
                matrix[i][j]=matrix[j][i];
                matrix[j][i]=temp;
            }
        }
        for(int i=0;i<n;i++){
            int left=0,right=n-1;
            while(left<right){
                int temp=matrix[i][left];
                matrix[i][left]=matrix[i][right];
                matrix[i][right]=temp;
                left++;
                right--;
            }
        }
    }
    // Function to print matrix
    public static void getPrintMatrix(int matrix[][]){
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix.length;j++){
                System.out.print(matrix[i][j]+" ");
            }
            System.out.println();
        }
    }
    // Main Method
    public static void main(String [] args){
        // Let's Start here
        int [][] matrix={
            {1,2,3},
            {4,5,6},
            {7,8,9}
        };
        System.out.println("Original Matrix : ");
        getPrintMatrix(matrix);
        System.out.println("Rotated Matrix : ");
        getRotate(matrix);
        getPrintMatrix(matrix);
    }
}

File: 100-Days-Series/Day 43/RotateMatrix90d.py

# Function to rotate a matrix by 90 degrees clockwise
def getRotate(matrix):
    for i in range(0,len(matrix),1):
        for j in range(i+1,len(matrix),1):
            temp=matrix[i][j]
            matrix[i][j]=matrix[j][i]
            matrix[j][i]=temp
    
    for i in range(0,len(matrix),1):
        left = 0
        right= len(matrix)-1
        while left<right:
            temp=matrix[i][left]
            matrix[i][left]=matrix[i][right]
            matrix[i][right]=temp
            left+=1
            right-=1




# Function for print matrix
def getPrintMatrix(matrix):
    for i in range(0,len(matrix),1):
        for j in range(0,len(matrix),1):
            print(matrix[i][j],end=" ")
        print()



# Let's Start here
matrix=[
    [1,2,3],
    [4,5,6],
    [7,8,9]
]
print('Original Matrix : ')
getPrintMatrix(matrix)
print('Rotated matrix : ')
getRotate(matrix)
getPrintMatrix(matrix)